improver/cli/generate_clearsky_solar_radiation.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2021 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to run GenerateClearSkySolarRadiation ancillary generation."""
+
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(
+    target_grid: cli.inputcube,
+    time: cli.inputdatetime,
+    accumulation_period: int,
+    *,
+    altitude: cli.inputcube = None,
+    linke_turbidity: cli.inputcube = None,
```

Comment:
I believe James Canvin is currently adding ancillaries to the graphing in your suite, which is the way it is set up in ours. This will likely mean that the ancillaries (like surface altitude) will be bundled together with the other inputs in an `{INPUTS}` argument when calling this CLI within the suite app. This essentially means you need to keep all the cubes together in your CLI interface as they get passed in as a group. 

There are quite a few examples of CLIs that have optional cubes grouped with the others, e.g. nbhood
```
def process(
    cube: cli.inputcube,
    mask: cli.inputcube = None,
    *,
    neighbourhood_output,
    ...
```
Can I suggest that you make all the arguments referring to cubes sequential for this reason.

Reply:
No worries. I've updated as requested.

I've also updated the interface in `generate_solar_time` to:
```
def process(target_grid: cli.inputcube, *, time: cli.inputdatetime):
```
so that time is passed from the command line in the same way as `generate_clearsky_solar_radiation` (and other similar interfaces).